lib/ansible/executor/task_executor.py

```diff
@@ -777,9 +777,13 @@ def _get_action_handler(self, connection, templar):
         '''
 
         module_prefix = self._task.action.split('_')[0]
+        metadata = self._shared_loader_obj.module_loader._load_module_metadata(
+                self._task.action, self._shared_loader_obj.module_loader.find_plugin(self._task.action))
 
         # let action plugin override module, fallback to 'normal' action plugin otherwise
-        if self._task.action in self._shared_loader_obj.action_loader:
+        if 'action_handler' in metadata:
+            handler_name = metadata['action_handler']
+        elif self._task.action in self._shared_loader_obj.action_loader:
```

Comment:
Not really, we want the module specific action handler to be the prevalent option.
The module_prefix check is left for now for already existing modules, but in the end I'm even thinking of modifying those volumes to have 'action_handler' on metadata and remove that special casing.